api/src/Controller/AuthController.php

```diff
@@ -43,21 +42,27 @@ public function login(
     ) {
         try {
             // See LoginRequestAuthenticator and RegistrationTokenAuthenticator for checks. User is set in token storage on successful authentication via Symfony event
-            /** @var User $user */
-            $user = $this->tokenStorage->getToken()->getUser();
+            $token = $this->tokenStorage->getToken();
 
-            $user->setLastLoggedIn(new \DateTime());
-            $em->persist($user);
-            $em->flush();
+            if (null !== $token) {
+                /** @var User $user */
+                $user = $token->getUser();
 
-            // Now doing this inline rather than injecting RedisUserProvider
-            $authToken = $user->getId().'_'.sha1(microtime().spl_object_hash($user).rand(1, 999));
-            $redis->set($authToken, serialize($this->tokenStorage->getToken()));
+                $user->setLastLoggedIn(new \DateTime());
+                $em->persist($user);
+                $em->flush();
 
-            // add token into response
-            $restInputOutputFormatter->addResponseModifier(function ($response) use ($authToken) {
-                $response->headers->set(HeaderTokenAuthenticator::HEADER_NAME, $authToken);
-            });
+                // Now doing this inline rather than injecting RedisUserProvider
+                $authToken = $user->getId().'_'.sha1(microtime().spl_object_hash($user).rand(1, 999));
```

Comment:
Is this just to generate a unique ID key for the token in Redis? Might be easier to use a uuid generator?

Reply:
Also, the user_id is a prefix on the redis key outside the hash? Might be better so be inside the hash if we never need to reference it to avoid any undue info leakage.

Reply:
Thanks for your feedback @mattmachell 
I've chatted to Jack and we're going to pull this into a separate tech debt ticket.